threesixty/views.py

```diff
@@ -290,10 +310,31 @@ def get_initial(self):
         return initial
 
     def form_valid(self, form):
-        if not self.survey.participant_can_skip and form.cleaned_data['decision'] is None:
+        if self.survey.participant_can_skip is False and form.cleaned_data['decision'] is None:
             return HttpResponseForbidden()
-        self.object = form.save(commit=False)
-        self.object.participant = self.participant
-        self.object.survey = self.survey
-        self.object.save()
-        return HttpResponseRedirect(self.request.path)
+        
+        if form.data['undo'] == 'true':
+            try:
+                latest_answer = Answer.objects.filter(participant=self.participant).latest('created')
+                kwargs = {
+                    'survey_pk': self.survey.pk,
+                    'token': self.token,
+                    'question_pk': latest_answer.question.pk
+                }
+                latest_answer.delete()
+                return HttpResponseRedirect(reverse('surver-answer-specific', kwargs=kwargs))
+            except Answer.DoesNotExist:
+                return self.redirect_survey_answer(self.survey.pk, self.token)
```

Comment:
Why? Who says that all questions are answered at this point? Why not go to `self.request.path`?

Reply:
Its the other way around. There are no more answeres, that means no question was answered yet and thats why we redirect to a random question.